flocker/control/_protocol.py

```diff
@@ -56,10 +54,10 @@ class DeploymentArgument(Argument):
     AMP argument that takes a ``Deployment`` object.
     """
     def fromString(self, in_bytes):
-        return deserialize_deployment(in_bytes)
+        return wire_decode(in_bytes)
 
     def toString(self, deployment):
-        return serialize_deployment(deployment)
+        return wire_encode(deployment)
```

Comment:
These two AMP types are pretty similar now.  Maybe even exactly the same.  Perhaps we only need one of them?  Or perhaps the way they should differ now is to disallow objects of the wrong type.